### PR TITLE
Switch JDK9 build to JDK11

### PR DESCRIPTION
**Pull request by ssoloff**
## Overview

Replaces the current Travis JDK9 build with a JDK11 build.

## Functional Changes

None.

## Manual Testing Performed

In addition to the Travis build, I ran the `release` target locally under JDK11 to sanity check it, since we haven't traditionally run that task for the JDK9 build. I smoke tested a local game using the resulting headed game portable build under both Java 8 and Java 11.

### CODE REVIEW

**Comment by ssoloff** [comment]
> I suppose the recent plugin updates have anything to do with it?

Yeah, I've been looking to do this for a while and have been slowly making changes to get us there.  The Error Prone Gradle plugin was the last stumbling block.

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ matrix:
   include:
   - jdk: oraclejdk8
     env: TRIPLEA_RELEASE=true
-  - jdk: oraclejdk9
+  - jdk: oraclejdk11
```
Comment (RoiEXLab) [inline .travis.yml]: A thing we might want to consider:
As of java 11 the Oracle JDK sort of lost its "de-facto"-JDK status, so we might want to switch to the openJDK due to the licensing restrictions.

**Comment by DanVanAtta** [comment]
@ssoloff any reason to not keep jdk9? I've found the multiple builds to be nice when exactly one fails, it helps to indicate a 'random' build failure, typically jenkins failing to download a jar dependency.

This may draw into question the bundled-jar problem, ideally we can enforce an official minimum java version. If we enforce a minimum then we likely should build that version and any above it. If we enforce an exact version then of course we could build exactly those.

File: .travis.yml (hunk 1 above)
Comment (DanVanAtta) [reply .travis.yml]: Second note, matching the version use for installer would be pretty slick.

**Comment by ssoloff** [comment]
> any reason to not keep jdk9?

It reached end-of-life ten months ago?

**Comment by DanVanAtta** [comment]
same could be said about java 8  ; )

I think any valid user configuration is a good candidate. To avoid JDK's moving the minimum enforced higher would be the best way to do it IMO. 

Are we trying to avoid the cost of building on more than two JDKs? Should JDK10 be introduced given users could use that configuration

**Comment by RoiEXLab** [comment]
@DanVanAtta Java 8 is still supported, and by no means end of life

**Comment by DanVanAtta** [comment]
Googling, seems JDK9 is not LTS:  https://www.oracle.com/technetwork/java/java-se-support-roadmap.html

8 and 11 are LTS. Seems that is the determining factor, perhaps 9 had no business being on the build list at all. In theory we should be trying very hard to test all environments/conditions that users would use.

**Comment by ssoloff** [comment]
> I'm surprised this works without any issues.

Actually, there is one issue: the JFX client no longer works because JFX was removed from the Oracle JDK as of 11, which puts it at parity with OpenJDK 11.  I remember talking about this when we added the `--release 8` argument for the compiler, so I'm pretty sure the fix is simply to change the `compileOnly` dependency on JFX to a `compile` dependency for the Java 11 build.  I'll fix that as soon as I get a chance, but I suspect it will dramatically increase the size of the release artifacts produced by the Java 11 build (if we ever choose to actually release them).

**Comment by ssoloff** [comment]
I partially retract my previous comment.  If we want to be able to run the JFX client on Java 11, then we have to bundle JFX with the _Java 8_ installers.  That would increase the installer sizes by ~18 MiB (compression notwithstanding).  I don't think that's a good idea at this time given that the JFX client is nowhere near being complete.  I'd suggest we wait to bundle JFX until either:

1. we drop support for Java 8, at which point we have no other choice, or
1. the JFX client supports a minimal feature set that makes it more usable (e.g. the ability to play a local game).

**Pull request merged**